src/write/xcoff.rs

```diff
@@ -0,0 +1,497 @@
+                        RelocationKind::Xcoff(x) => x,
+                        _ => {
+                            return Err(Error(format!("unimplemented relocation {:?}", reloc)));
+                        }
+                    };
+                    if is_64 {
+                        let xcoff_rel = xcoff::Rel64 {
+                            r_vaddr: U64::new(BE, reloc.offset as u64),
+                            r_symndx: U32::new(BE, symbol_offsets[reloc.symbol.0].index as u32),
+                            // Specifies the bit length of the relocatable reference minus one.
+                            r_rsize: (reloc.size - 1),
+                            r_rtype: rtype,
+                        };
+                        buffer.write(&xcoff_rel);
+                    } else {
+                        let xcoff_rel = xcoff::Rel32 {
+                            r_vaddr: U32::new(BE, reloc.offset as u32),
+                            r_symndx: U32::new(BE, symbol_offsets[reloc.symbol.0].index as u32),
+                            r_rsize: (reloc.size - 1),
+                            r_rtype: rtype,
+                        };
+                        buffer.write(&xcoff_rel);
+                    }
+                }
+            }
+        }
+
+        // Write symbols.
+        debug_assert_eq!(symtab_offset, buffer.len());
+        for (index, symbol) in self.symbols.iter().enumerate() {
+            let (section_number, section) = match symbol.section {
+                SymbolSection::None => {
+                    debug_assert_eq!(symbol.kind, SymbolKind::File);
+                    (xcoff::N_DEBUG, None)
+                }
+                SymbolSection::Undefined | SymbolSection::Common => (xcoff::N_UNDEF, None),
+                SymbolSection::Absolute => (xcoff::N_ABS, None),
+                SymbolSection::Section(id) => (id.0 as i16 + 1, Some(&self.sections[id.0])),
+            };
+            let storage_class = match symbol.kind {
+                SymbolKind::File => xcoff::C_FILE,
+                SymbolKind::Null => xcoff::C_NULL,
+                SymbolKind::Data => xcoff::C_STAT,
+                SymbolKind::Label => {
+                    if symbol.is_undefined() {
+                        xcoff::C_ULABEL
+                    } else {
+                        xcoff::C_LABEL
+                    }
+                }
+                SymbolKind::Tls => {
+                    if symbol.is_local() {
+                        xcoff::C_STTLS
+                    } else {
+                        xcoff::C_GTLS
+                    }
+                }
+                SymbolKind::Section => {
+                    if symbol.weak {
+                        xcoff::C_WEAKEXT
+                    } else if symbol.is_undefined() {
+                        xcoff::C_HIDEXT
+                    } else {
+                        xcoff::C_EXT
+                    }
+                }
+                SymbolKind::Text => {
```

Comment:
This is the same as `SymbolKind::Data`. It should be `C_STAT` or `C_EXT` depending on `symbol.is_local()`. It should never be `C_FUN` (which is a debugging symbol) or `C_INFO` (which is not text).

For `C_INFO`, I think we should add `SymbolFlags::Xcoff` and allow the user to request `C_INFO` via that.